test/pylib/scylla_cluster.py

```diff
@@ -518,10 +517,10 @@ async def cql_is_up(self) -> CqlUpState:
                                                    control_connection_timeout=self.TOPOLOGY_TIMEOUT,
                                                    auth_provider=auth)
                     self.control_connection = self.control_cluster.connect()
-                    return CqlUpState.QUERIED
+                    return ServerState.CQL_QUERIED
         except (NoHostAvailable, InvalidRequest, OperationTimedOut) as exc:
             self.logger.debug("Exception when checking if CQL is up: %s", exc)
-            return CqlUpState.CONNECTED if connected else CqlUpState.NOT_CONNECTED
+            return ServerState.CONNECTED if connected else ServerState.HOST_ID_REQUIRED
```

Comment:
![Screenshot_20240821_192502](https://github.com/user-attachments/assets/d126bb6b-771b-4994-ac86-04172fbb151c)

Reply:
Ok, but let's make the naming consistent, accurate, and remove abstraction leak:
- `HOST_ID_REQUIRED` should be `HOST_ID_QUERIED`. This is the state we enter **after** querying host ID. (see also my other comments below where the states are assigned) "host ID required" is really the same as "process started" so we don't need it
- I'd rename `get_server_state` back to `cql_is_up` because this function really is only dealing with CQL connectivity.
- rename `ServerState` -> `ServerUpState` to be more precise
- rename `CONNECTED` -> `CQL_CONNECTED` to be more precise
- now `cql_is_up` should not be aware of "host ID", the fact that it returns `HOST_ID_<something>` in some cases is an abstraction leak. To get rid of it, I'd introduce a parameter to this function, `start_state`, and if we fail to connect, return `start_state`. At the callsite we would pass `start_state` to be the state we're currently in (it will be `HOST_ID_REQUIRED`) (see comment below)
- alternatively if you don't like passing `start_state` to `cql_is_up`, we could make `cql_is_up` return `ServerUpState | None` -- if it fails to connect, return `None`. Then at call site we would do `server_up_state = await self.cql_is_up() or server_up_state` (which means: if we successfully connect with CQL, assign the new state after connecting; if we fail, we remain in whatever state we had (which would be `HOST_ID_REQUIRED`)). I think I actually like this idea more than passing `start_state`, but choose your favorite

Reply:
OK, will go with your latest idea. One thing to improve naming here is to use name `get_cql_up_state` instead of `cql_is_up` which sounds like it should return boolean value.